src/bindgen/mangle.rs

```diff
@@ -43,16 +49,15 @@ impl<'a> Mangler<'a> {
     }
 
     fn push(&mut self, id: Separator) {
-        let separator = '_';
         let count = id as usize;
-        self.output.extend(std::iter::repeat(separator).take(count));
+        self.output.extend(std::iter::repeat(self.mangle_separator).take(count));
     }
 
     fn append_mangled_type(&mut self, ty: &Type, last: bool) {
         match *ty {
             Type::Path(ref generic) => {
                 let sub_path =
-                    Mangler::new(generic.export_name(), generic.generics(), last).mangle();
+                    Mangler::new(generic.export_name(), generic.generics(), last, &Some(self.mangle_separator.to_owned())).mangle();
```

Comment:
This should not need to allocate, The function should take an `Option<&'a str>` instead, and this should just be `Some(self.mangle_separator)`.

Reply:
Agreed, I tried to make it all `Option<&str>` but it got tricky since `Config::export` stores `Option<String>`. I now see that that we can convert to `Option<&str>` with `Option::as_deref` so I'll get that sorted.

Reply:
Fixed with 7f5b606 (plus some tidy ups)